server/services/forms/schemas/add-correspondent-details.js

```diff
@@ -1,6 +1,14 @@
 const Form = require('../form-builder');
 const { Component, Choice } = require('../component-builder');
 
+function buildUrl(options) {
+    if (options.action === 'addNoMp') {
+        return `/case/${options.caseId}/stage/${options.stageId}`;
+    } else {
+        return `/case/${options.caseId}/stage/${options.stageId}/entity/correspondent/add`;
+    }
+}
```

Comment:
Slightly cleaner like:
```suggestion
function buildUrl(options) {
    let url = `/case/${options.caseId}/stage/${options.stageId}`;

    if (options.action === 'addNoMp') {
        return url;
    }

    return url + `/entity/correspondent/add`;
}
```

Could also add tests too

Reply:
That `let` can be a `const`.

Reply:
Indeed it can, fixed :)